In [1]:
#problem 1
# Solution :

As stated in Problem  , 

The mean of population μ (mu)= 100

The standard deviation of problem σ (sigma) = 15

In order to prove that ,  raw cornstarch had an effect or not on blood glucose level ,

Hypothesis test , 

Stpe 1: 

( Null Hypothesis : That diet of raw cornstartch has effect upon Blood glucose level ) H0: μ =100

( Alternative Hypothesis : That diet of raw cornstartch has no effect upon Blood glucose level ) H1: μ >100


Step 2:

Significance level (α) , , which refers to the degree of significance in which we accept or reject the null-hypothesis.
Generally 100% accuracy is not possible for accepting or rejecting a hypothesis.
Hence consider significance level α =5% =0.05



Step 3:

 
Hypothesis testing can be done through many ways , one of which is Z-test (whcih predict the standard, normal distribution of data). A Z-test is any statistical test, for which the distribution of the test statistic under the null hypothesis can be approximated by a normal distribution. Z- test , return the critical value (or region of rejection) which decide that nul hypothesis can be rejected or not.

Formula: 

       Z = (Sample mean (X) - Population mean (μ ) )/ ( Population Standard deviation ( σ)/ sqrt(No. Of Samples(N)) )



In [23]:
# Calculation 

μ =100 # mean of population
X= 108 # mean of sample for 36 samples
σ= 15 # standard deviation  of population
N = 36 # No. Of Samples 




import math   

import scipy.stats as stats 

Z = (X - μ) / ((σ) /math.sqrt(N))
print(" Z-Score value is :",Z)

 Z-Score value is : 3.2


Note :cdf  is Cumulative density function , is the probability that the variable takes a value less than or equal to x. 
That is F(x)=Pr[X≤x]=α

In [24]:


p = stats.norm.cdf(Z)  
print("The probability of having mean less than 108 ( mean as per sample) is ",round(p,4))

The probability of having mean less than 108 ( mean as per sample) is  0.9993


As per sample , 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108 . Hence probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.

----------------------------------------------------------------------------------------------------------------------------

In [20]:
#problem 2
# Solution 

Probability ananlysis:

Let consider that,

Probability of having Republican voters in First state P1 =52% =0.52

Probability of having Republican voters in Second state P2 =47% =0.47


p1 is the probability for  proportion of Republican voters in the sample from the first state

p2 is the probability for  proportion of Republican voters in the sample from the Second state

Number of voters sampled from First state n1 =100

Number of voters sampled from Second state n2 =100


In order to calculation, the sample size should be large enough  to model differences with a normal population, 
Hence , 

Sample of republican voters from first state > n1P1 = 100 * 0.52 = 52, 

Sample of democratic voters from first state > n1(1 - P1) = 100 * 0.48 = 48, 

Sample of republican voters from Second state > n2P2 = 100 * 0.47 = 47, 

Sample of democratic voters from Second state >  n2(1 - P2) = 100 * 0.53 = 53 

In [39]:
# Calculate mean of the difference in sample proportions



P1 = 0.52
P2= 0.47
μ = P1 - P2

print("The mean(μ) of sample proportions is ", round(μ,2))  # round funtion to get values after decimal

The mean(μ) of sample proportions is  0.05


In [35]:
#Calculate Standard deviation (σ) of the difference

# P1 = Probability of having Republican voters in First state =0.52
# 1-P1 = Probability of having democratic voters in First state =0.48
# P2 = Probability of having democratic voters in First state =0.47
# 1-P2 = Probability of having democratic voters in First state =0.53

# n1 = No. of samples from first state=100
# n2 = No. of samples from Second  state=100

P1=0.52
P2=0.47
n1=100
n2=100

# import python module

import math  # for mathematical calculation
σ = math.sqrt(( P1*(1 - P1) / n1 ) + ( P2*(1 - P2) / n2 ) )
print("The standard deviation of difference is",σ)

The standard deviation of difference is 0.07061869440877536


In order to calculate that , p1 is less than p2 
where , 
p1 =  is the probability for proportion of Republican voters in the sample from the first state
p2 =  is the probability for proportion of Republican voters in the sample from the Second state

Hence , to calculate p1 is less than p2 , which is equal to finding the probability that p1 - p2 is less than zero.
Thus standard normal probablity ( z-score) for random variable (p1-p2) is :

Z = ( sample mean of differnce (X) -  sample mean of differnce (μ) ) / ( Standard devaition of difference (σ) )


In [44]:
X = 0  # as  , mean difference of samples p1 - p2 is less than zero
μ # mean of sample proportions or sample mean of difference
σ # standard deviation of difference in population proportions


Z = (X -μ)/σ
print("The Z-Score is:" , Z)

The Z-Score is: -0.7080278164104213


In [49]:

import scipy.stats as stats # python module for statistical calculations

# calcultion of standard normal probabilities based upon " Z - Score"

p = stats.norm.cdf(Z)  # cdf function takes Z- score , based upon which return the standard normal probality
print("Output:\n")
print("The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is ",round(p,2))

Output:

The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is  0.24


---------------------------------------------------------------------------------------------------------------------------

<b> Problem 3</b> You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

In [52]:
#problem 3
# Solution 


# mean of SAT score is μ = 1026
# Standard deviation of SAT score is σ  = 209



X =1100
σ = 209
μ = 1026
# Z-score equation : (sample mean - population mean)/ (standard mean of error( which is equal to standard devaition/square root of no. of samples))
Z = (X-μ )/σ 
print("The Z-Score value is",round(Z,4))

The Z-Score value is 0.3541


In [58]:
import scipy.stats as stats 

p = stats.norm.cdf(Z) 
print("Output:\n")
print("The probabiltiy that Student's score on the test compared to the average test taker is",round(p,4) , 'or' , round((p*100),2),'%')

Output:

The probabiltiy that Student's score on the test compared to the average test taker is 0.6384 or 63.84 %
